### PR TITLE
bpo-29727: Register array.array as a MutableSequence

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue29727

### CODE REVIEW

File: Modules/arraymodule.c [hunk 1]
```diff
@@ -2989,6 +2989,26 @@ array_modexec(PyObject *m)
...
+        Py_DECREF((PyObject *)&Arraytype);
+        return -1;
+    }
+    PyObject *mutablesequence = PyObject_GetAttrString(abc_mod, "MutableSequence");
+    Py_DECREF(abc_mod);
+    if (!mutablesequence) {
+        Py_DECREF((PyObject *)&Arraytype);
+        return -1;
+    }
+    PyObject *res = PyObject_CallMethod(mutablesequence, "register", "O", (PyObject *)&Arraytype);
```
Comment (brandtbucher) [inline Modules/arraymodule.c:3004]:
```suggestion
    _Py_IDENTIFIER(register);
    PyObject *res = _PyObject_CallMethodIdOneArg(mutablesequence, &PyId_register,
                                                 (PyObject *)&Arraytype);
```

**Review by brandtbucher — APPROVED** [review]
I like the tests! Noticed one small opportunity to simplify the `register` call, but looks good either way:

File: Modules/arraymodule.c (hunk 1 above)
Comment (pablogsal) [reply Modules/arraymodule.c:3004]: Thanks, @brandtbucher. The reason I originally did not add `_Py_IDENTIFIER` is so we don't add more to the global/static state so it does not make future sub interpreters isolation harder, especially given that this code will run only at module import time.

**Comment by pablogsal** [comment]
Thanks for the review!

**Comment by pablogsal** [comment]
> Hi! The buildbot **aarch64 RHEL7 3.x** has failed when building commit [e51dd9d](https://github.com/python/cpython/commit/e51dd9dad6590bf3a940723fbbaaf4f64a3c9228).

The failure is unrelated to this PR:

> 1 test altered the execution environment:
> test_asyncio

**Pull request merged**